migration-tests/pallets/vesting.ts

```diff
@@ -0,0 +1,97 @@
+import '@polkadot/api-augment';
+import assert from 'assert';
+import { PreCheckContext, PostCheckContext, MigrationTest, PreCheckResult } from '../types.js';
+import type { Codec } from '@polkadot/types/types';
+import type { StorageKey } from '@polkadot/types';
+
+export const vestingTests: MigrationTest = {
+    name: 'vesting_pallet',
+    pre_check: async (context: PreCheckContext): Promise<PreCheckResult> => {
+        const { rc_api_before, ah_api_before } = context;
+
+        // Collect RC data
+        const rc_vestingEntries = await rc_api_before.query.vesting.vesting.entries();
+        const rc_storageVersion = await rc_api_before.query.vesting.storageVersion();
+
+        // AH Pre-check assertions
+        const ah_vestingStorageVersion = await ah_api_before.query.vesting.storageVersion();
+        assert.equal(
+            ah_vestingStorageVersion.toHuman(),
+            'V0',
+            'Vesting storage version should be V0 before migration'
+        );
+
+        const ah_vestingEntries = await ah_api_before.query.vesting.vesting([]);
+        assert(
+            ah_vestingEntries.isEmpty,
+            'Vesting entries before migration should be empty'
+        );
+
+        return {
+            rc_pre_payload: {
+                vestingEntries: rc_vestingEntries,
+                storageVersion: rc_storageVersion
+            },
+            ah_pre_payload: undefined
+        };
+    },
+
+    post_check: async (
+        context: PostCheckContext,
+        pre_payload: PreCheckResult
+    ): Promise<void> => {
+        const { rc_api_after, ah_api_after } = context;
+        const { vestingEntries: rc_vestingEntries_before } = pre_payload.rc_pre_payload;
+
+        // Check RC is empty after migration
+        const rc_vestingEntries_after = await rc_api_after.query.vesting.vesting([]);
+        assert(
+            rc_vestingEntries_after.isEmpty,
+            'Vesting entries after migration should be empty'
+        );
+
+        // Get AH entries after migration
+        const ah_vestingEntries_after = await ah_api_after.query.vesting.vesting.entries();
```

Comment:
This is not a concern whether at this point in time or for this pallet in particular, but it would be good to keep in mind that these unbounded queries can quickly grow into several MBs in size for some pallet/storage pairs.

As we prepare to test on Kusama and then Polkadot, introducing optional paging for these tests is an option.